projects/core/utils/mask/create-number-mask.ts

```diff
@@ -108,14 +108,29 @@ export function tuiCreateNumberMask({
             mask.unshift('-');
         }
 
-        return preventLeadingZeroes(mask);
+        return preventLeadingZeroes(mask, thousandSymbol, integerCappedClean);
+        //return mask;
     };
 }
 
-function preventLeadingZeroes(mask: Array<string | RegExp>): Array<string | RegExp> {
+function preventLeadingZeroes(
+    mask: Array<string | RegExp>,
+    thousandSymbol: string = CHAR_NO_BREAK_SPACE,
+    integerCappedClean: string,
+): Array<string | RegExp> {
     const firstDigitIndex = mask.indexOf(TUI_DIGIT_REGEXP);
 
-    if (firstDigitIndex !== -1 && mask[firstDigitIndex + 1] === TUI_DIGIT_REGEXP) {
+    if (firstDigitIndex === -1) {
+        return mask;
+    }
+
+    const isOnlyNonZeroDigit =
+        mask.length === 1 && NON_ZERO_DIGIT.test(integerCappedClean);
+
+    if (
+        isOnlyNonZeroDigit ||
+        [TUI_DIGIT_REGEXP, thousandSymbol].includes(mask[firstDigitIndex + 1])
```

Comment:
We do not use `includes` on arrays, in this major version we stick to es6 code. When we bump supported browser versions in 3.0, I think we will start using it, but for now we stick to `indexOf`.

Reply:
refactored this part